testnet/stacks-node/src/nakamoto_node/miner.rs

```diff
@@ -396,6 +409,10 @@ impl BlockMinerThread {
         let now = Instant::now();
         debug!("Miner: waiting for block response from reward cycle {reward_cycle } signers...");
         while now.elapsed() < self.config.miner.wait_on_signers {
+            if self.check_burn_tip_changed(&sortdb).is_err() {
```

Comment:
This check isn't sufficient.  While waiting for a block, the Bitcoin chain can "flap" between tips A, B, and back to A again.  Then, this check could pass (i.e. return `Ok(..)`) if the flap happens during one pass of the loop, but the signers could notice the flap and produce two conflicting Stacks histories (one for each Bitcoin fork). However, it could be the case that tips A and B have no sortition, but they share a common ancestor that does (so, the conflicting histories' tenure-change transactions would be consistent, but their in-flight tenures would diverge).  To address this, I think this code should additionally check that the Stacks tip hasn't changed, except to add the miner's in-flight block.

Reply:
I agree, but I think the miner should just continue trying to get its block accepted up to the configured timeout with signers. If the signer set has moved onto another miner, it will be realized in a timeout, and I don't think that the logic needs to be more complicated than that.